jsonargparse/typehints.py

```diff
@@ -410,7 +410,16 @@ def adapt_typehints(val, typehint, serialize=False, instantiate_classes=False, s
                 cast = str if serialize else int
                 val = {cast(k): v for k, v in val.items()}
             for k, v in val.items():
-                val[k] = adapt_typehints(v, subtypehints[1], **adapt_kwargs)
+                if "linked_targets" in adapt_kwargs["sub_add_kwargs"]:
+                    kwargs = adapt_kwargs.copy()
+                    sub_add_kwargs = kwargs["sub_add_kwargs"]
+                    sub_add_kwargs["linked_targets"] = {t[len(k + "."):] for t in sub_add_kwargs["linked_targets"]
+                                                        if t.startswith(k + ".")}
+                    sub_add_kwargs["linked_targets"] = {t[len("init_args."):] if t.startswith("init_args.") else t
+                                                        for t in sub_add_kwargs["linked_targets"]}
```

Comment:
I see that now there is a unit test. However these lines are not covered by the test. Have a look at https://codecov.io/gh/omni-us/jsonargparse/src/32e125a857f7a2888ab1db063862d527e98ed7fc/jsonargparse/typehints.py#